### PR TITLE
react-auth: build with parcel, add CI job (GEA-12703)

### PR DESCRIPTION
**Pull request by kenany**
Verified working locally in pangea-integration-nextjs-authn.

Also verified that this fixes the package for the internal platform/docs app, which is currently not using this package because of a build error.

### CODE REVIEW

File: packages/react-auth/package.json [hunk 1]
```diff
@@ -2,20 +2,24 @@
...
+    "dist"
+  ],
   "repository": "https://github.com/pangeacyber/pangea-javascript.git",
   "author": "Pangea (https://pangea.cloud)",
   "license": "MIT",
   "scripts": {
-    "build": "rm -rf dist && rollup -c",
-    "lint": "npx prettier --check .",
-    "prettier:fix": "npx prettier . -w"
+    "build": "parcel build",
```
Comment (cj-f) [inline packages/react-auth/package.json:16]: Is the idea to switch every package from rollup to parcel? Just curious what was the main reason? Parcel being zero config is just easier and less error prone?

**Review by cj-f — APPROVED** [review]
Approving, although @ggallien13 should review as well, as he primarily works with this package

File: packages/react-auth/package.json (hunk 1 above)
Comment (ggallien13) [reply packages/react-auth/package.json:16]: I have the same question, what's the reason for the change?

**Review by ggallien13 — DISMISSED** [review]

File: packages/react-auth/package.json (hunk 1 above)
Comment (kenany) [reply packages/react-auth/package.json:16]: Yeah I went with this in part to try Parcel but more importantly to simplify the current setup, which had been stuck on Rollup v3 for 2 years. I'm hoping with this there will be less packages to keep updated and less config to manage.

Brings this to the other packages would probably be nice but I don't have my sights set on that just yet.

**Comment by kenany** [comment]
Rebased onto latest `main` and resolved merge conflicts.

**Pull request merged**